flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt

```diff
@@ -6,94 +6,163 @@
+        }
+        widgetsTree.selectRow(6) // Text widget
+        pause(object : Condition("Details tree changes") {
+          override fun test(): Boolean {
+            return initialDetails != detailsTree.selectionSync().toString()
+          }
+        }, Timeouts.seconds05)
+        val row: JTreeRowFixture = detailsTree.treeFixture().node(1)
+        val expected = "\"You have pushed the button that many times:\""
+        expect(expected) { row.value() }
+      }
+
+      runner().stop()
+    }
   }
 
   fun IdeFrameFixture.launchFlutterApp() {
-    findRunApplicationButton().click()
-    val runner = runner()
-    pause(object : Condition("Start app") {
-      override fun test(): Boolean {
-        return runner.isExecutionInProgress
-      }
-    }, Timeouts.seconds10)
+    step("Launch Flutter app") {
+      findRunApplicationButton().click()
+      val runner = runner()
+      pause(object : Condition("Start app") {
+        override fun test(): Boolean {
+          return runner.isExecutionInProgress
+        }
+      }, Timeouts.seconds30)
+    }
   }
 
-  fun IdeFrameFixture.selectSimulator() {
-    selectDevice("Open iOS Simulator")
+  private fun findHotReloadButton(): ActionButtonFixture {
+    return findActionButtonByClassName("ReloadFlutterAppRetarget")
   }
 
-  fun IdeFrameFixture.selectDevice(devName: String) {
-    val runButton = findRunApplicationButton()
-    val actionToolbarContainer = execute<Container>(object : GuiQuery<Container>() {
-      @Throws(Throwable::class)
-      override fun executeInEDT(): Container? {
-        return runButton.target().parent
-      }
-    })
-    assertNotNull(actionToolbarContainer)
-
-    // These next two lines look like the right way to select the simulator, but it does not work.
-    //    val comboBoxActionFixture = ComboBoxActionFixture.findComboBoxByText(robot(), actionToolbarContainer!!, "<no devices>")
-    //    comboBoxActionFixture.selectItem(devName)
-    // Need to get focus on the combo box but the ComboBoxActionFixture.click() method is private, so it is inlined here.
-    val selector = button("<no devices>")
-    val comboBoxButtonFixture = JButtonFixture(robot(), selector.target())
-    GuiTestUtilKt.waitUntil("ComboBoxButton will be enabled", Timeouts.seconds10) {
-      GuiTestUtilKt.computeOnEdt { comboBoxButtonFixture.target().isEnabled } ?: false
-    }
-    comboBoxButtonFixture.click()
-    robot().pressAndReleaseKey(KeyEvent.VK_DOWN)
-    robot().pressAndReleaseKey(KeyEvent.VK_ENTER)
-    robot().waitForIdle()
+  private fun IdeFrameFixture.runner(): ExecutionToolWindowFixture.ContentFixture {
+    return runToolWindow.findContent("main.dart")
   }
 
-  private fun IdeFrameFixture.findActionButtonByActionId(actionId: String): ActionButtonFixture {
+  private fun findActionButtonByActionId(actionId: String): ActionButtonFixture {
+    // This seems to be broken, but finding by simple class name works.
```

Comment:
Bummer.  But I guess since we own all the classes under test, not a big deal?

Reply:
I'm not sure what the problem is. I haven't even ruled out pilot error (hmm, that may not be a great idiom these days). It might be a bug in the framework. I keep this around to remind myself not to use it.